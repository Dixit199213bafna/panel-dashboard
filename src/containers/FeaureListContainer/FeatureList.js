import React, { Component } from 'react';
import classes from './FeatureList.module.css';
import { Accordian, Feature } from "../../components";
import panelList from "../../mock/panel-list.mock";
import PANEL_TYPE from '../../const/panel.const';

class FeatureList extends Component {
    constructor() {
        super();
        this.state = {
            panelList
        }
    }

    onValueChange(event) {
        this.setState({
            panelList: {
                ...this.state.panelList,
                [event.panel]: {
                    ...this.state.panelList[event.panel],
                    featureList: this.state.panelList[event.panel].featureList.map((feature) => {
                        if (event.childPanelId && feature.featureId === event.childPanelId) {
                            feature = {
                                ...feature,
                                childFeatureList: feature.childFeatureList.map((childFeat) => {
                                    if (childFeat.featureId === event.featureId) {
                                        childFeat.value = parseInt(event.value)
                                    }
                                    return childFeat;
                                })
                            }
                        }
                        if (feature.featureId === event.featureId) {
                            feature.value = parseInt(event.value);
                        }
                        return feature;
                    })
                }
            }
        })
    }

    onToggleClick(event) {
        this.setState({
            panelList: {
                ...this.state.panelList,
                [event.panel]: {
                    ...this.state.panelList[event.panel],
                    featureList: this.state.panelList[event.panel].featureList.map((feature) => {
                        if (event.childPanelId && feature.featureId === event.childPanelId) {
                            feature = {
                                ...feature,
                                childFeatureList: feature.childFeatureList.map((childFeat) => {
                                    if (childFeat.featureId === event.featureId) {
                                        childFeat.toggle = event.value
                                    }
                                    return childFeat;
                                })
                            }
                        }
                        if (feature.featureId === event.featureId) {
                            feature.toggle = event.value;
                        }
                        if (feature.featureId === event.featureId && feature.childFeatureList) {
                            feature.childFeatureList = feature.childFeatureList.map(feat => ({ ...feat, toggle: event.value }))
                        }
                        return feature;
                    })
                }
            }
        })
    }
    render() {
        const htmlContent = Object.keys(this.state.panelList).map((panel, index) => {
            return (
                <React.Fragment key={this.state.panelList[panel].panelId}>
                    <p>{panel}</p>
                    <div className={this.state.panelList[panel].type === PANEL_TYPE.INDIVIDUAL ? classes.Grid : null}>
                        {this.state.panelList[panel].featureList.map((feature) => feature.childFeatureList ? <Accordian panel={panel} onHeaderToggleClick={(e) => this.onToggleClick(e)} onChildValueChange={(e) => this.onValueChange(e)} onFeatureChildToggleClick={(e) => this.onToggleClick(e)} key={feature.featureId} {...feature} /> : <Feature onValueChange={(e) => this.onValueChange(e)} onToggleClick={(e) => this.onToggleClick(e)} key={feature.featureId} {...feature} panel={panel} />)}
                    </div>
                </React.Fragment>
            )
        });
        return htmlContent
    }
}

export default FeatureList