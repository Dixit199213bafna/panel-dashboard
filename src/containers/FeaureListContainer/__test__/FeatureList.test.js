import React from 'react';
import renderer from 'react-test-renderer';
import FeatureList from '../FeatureList';
import TestRenderer from 'react-test-renderer';
const {act} = TestRenderer;

describe('FeatureList', () => {
  let tree;
  const component = renderer.create(
    <FeatureList/>,
  );
  const instance = component.getInstance();

  it('FeaureList Initialized', () => {
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })
  
  it('FeaureList:Individual:On Toggle Click', () => {
    instance.onToggleClick({
        panel: 'General',
        featureId: 11,
        value: true
    });
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('FeaureList: Value Change', () => {
    instance.onValueChange({
        panel: 'Alerts',
        featureId: 32,
        value: 5
    });
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('FeatureList::Group:On Toggle Click', async () => {
    act(() => {
        component.root.findByProps({ className: "toggle" }).props.onClick();
    });
    instance.onToggleClick({
        panel: 'Settings',
        featureId: 22,
        value: true
    });
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })
});