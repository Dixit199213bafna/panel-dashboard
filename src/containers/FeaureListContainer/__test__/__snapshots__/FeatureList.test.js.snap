// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeatureList FeatureList::Group:On Toggle Click 1`] = `
Array [
  <p>
    General
  </p>,
  <div
    className="Grid"
  >
    <div
      className="Feature"
    >
      <label>
        <span
          className="GreyBox"
        />
        <span>
          CASE MANAGEMENT
        </span>
      </label>
      <div
        className="ToggleWithMax"
      >
        <div
          className="form-switch"
        >
          <input
            checked={true}
            className="form-check-input"
            disabled={false}
            id={11}
            onChange={[Function]}
            type="checkbox"
          />
        </div>
      </div>
    </div>
    <div
      className="Feature"
    >
      <label>
        <span
          className="GreyBox"
        />
        <span>
          MAP TIMELINE
        </span>
      </label>
      <div
        className="ToggleWithMax"
      >
        <div
          className="form-switch"
        >
          <input
            checked={false}
            className="form-check-input"
            disabled={false}
            id={12}
            onChange={[Function]}
            type="checkbox"
          />
        </div>
      </div>
    </div>
    <div
      className="Feature"
    >
      <label>
        <span
          className="GreyBox"
        />
        <span>
          VIEWS AND BRIEFINGS
        </span>
      </label>
      <div
        className="ToggleWithMax"
      >
        <div
          className="form-switch"
        >
          <input
            checked={false}
            className="form-check-input"
            disabled={false}
            id={13}
            onChange={[Function]}
            type="checkbox"
          />
        </div>
      </div>
    </div>
    <div
      className="Feature"
    >
      <label>
        <span
          className="GreyBox"
        />
        <span>
          NOTIFICATIONS
        </span>
      </label>
      <div
        className="ToggleWithMax"
      >
        <div
          className="form-switch"
        >
          <input
            checked={false}
            className="form-check-input"
            disabled={false}
            id={14}
            onChange={[Function]}
            type="checkbox"
          />
        </div>
      </div>
    </div>
    <div
      className="Feature"
    >
      <label>
        <span
          className="GreyBox"
        />
        <span>
          MASS COMMUNICATION
        </span>
      </label>
      <div
        className="ToggleWithMax"
      >
        <div
          className="form-switch"
        >
          <input
            checked={false}
            className="form-check-input"
            disabled={false}
            id={15}
            onChange={[Function]}
            type="checkbox"
          />
        </div>
      </div>
    </div>
    <div
      className="Feature"
    >
      <label>
        <span
          className="GreyBox"
        />
        <span>
          TRAFIC CAMERAS
        </span>
      </label>
      <div
        className="ToggleWithMax"
      >
        <div
          className="form-switch"
        >
          <input
            checked={false}
            className="form-check-input"
            disabled={false}
            id={16}
            onChange={[Function]}
            type="checkbox"
          />
        </div>
      </div>
    </div>
  </div>,
  <p>
    Settings
  </p>,
  <div
    className={null}
  >
    <div
      className="Feature"
    >
      <label>
        <span
          className="GreyBox"
        />
        <span>
          AUDIT LOG
        </span>
      </label>
      <div
        className="ToggleWithMax"
      >
        <div
          className="form-switch"
        >
          <input
            checked={false}
            className="form-check-input"
            disabled={false}
            id={21}
            onChange={[Function]}
            type="checkbox"
          />
        </div>
      </div>
    </div>
    <div
      className="accordion-item"
    >
      <div
        className="AccordianTitle"
      >
        <div
          className="FeatureAccordionHeader"
        >
          <label>
            <span
              className="GreyBox"
            />
            <span>
              USERS
            </span>
          </label>
          <div
            className="ToggleWithMax"
          >
            <div
              className="form-switch"
            >
              <input
                checked={true}
                className="form-check-input"
                disabled={false}
                id={22}
                onChange={[Function]}
                type="checkbox"
              />
            </div>
          </div>
        </div>
        <div
          className="toggle"
          onClick={[Function]}
        >
          <span
            className="DownCaret"
          />
        </div>
      </div>
      <div
        className="AccordionContent"
      >
        <div
          className="FeatureAccordionHeader"
        >
          <label>
            <span
              className="GreyBox"
            />
            <span>
              USERS - Add
            </span>
          </label>
          <div
            className="ToggleWithMax"
          >
            <div
              className="form-switch"
            >
              <input
                checked={true}
                className="form-check-input"
                disabled={false}
                id={221}
                onChange={[Function]}
                type="checkbox"
              />
            </div>
          </div>
        </div>
        <div
          className="FeatureAccordionHeader"
        >
          <label>
            <span
              className="GreyBox"
            />
            <span>
              USERS - Edit
            </span>
          </label>
          <div
            className="ToggleWithMax"
          >
            <div
              className="form-switch"
            >
              <input
                checked={true}
                className="form-check-input"
                disabled={false}
                id={222}
                onChange={[Function]}
                type="checkbox"
              />
            </div>
          </div>
        </div>
        <div
          className="FeatureAccordionHeader"
        >
          <label>
            <span
              className="GreyBox"
            />
            <span>
              USERS - Delete
            </span>
          </label>
          <div
            className="ToggleWithMax"
          >
            <div
              className="form-switch"
            >
              <input
                checked={true}
                className="form-check-input"
                disabled={false}
                id={223}
                onChange={[Function]}
                type="checkbox"
              />
            </div>
          </div>
        </div>
        <div
          className="FeatureAccordionHeader"
        >
          <label>
            <span
              className="GreyBox"
            />
            <span>
              Max USERS
            </span>
          </label>
          <div
            className="ToggleWithMax"
          >
            <select
              className="Select"
              onChange={[Function]}
              value={5}
            >
              <option
                value={5}
              >
                5
              </option>
              <option
                value={10}
              >
                10
              </option>
              <option
                value={20}
              >
                20
              </option>
              <option
                value={50}
              >
                50
              </option>
            </select>
            <div
              className="form-switch"
            >
              <input
                checked={true}
                className="form-check-input"
                disabled={false}
                id={224}
                onChange={[Function]}
                type="checkbox"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  <p>
    Alerts
  </p>,
  <div
    className={null}
  >
    <div
      className="Feature"
    >
      <label>
        <span
          className="GreyBox"
        />
        <span>
          ALERTS MANAGER
        </span>
      </label>
      <div
        className="ToggleWithMax"
      >
        <div
          className="form-switch"
        >
          <input
            checked={false}
            className="form-check-input"
            disabled={false}
            id={31}
            onChange={[Function]}
            type="checkbox"
          />
        </div>
      </div>
    </div>
    <div
      className="Feature"
    >
      <label>
        <span
          className="GreyBox"
        />
        <span>
          ALERT RULES
        </span>
      </label>
      <div
        className="ToggleWithMax"
      >
        <select
          className="Select"
          onChange={[Function]}
          value={5}
        >
          <option
            value={5}
          >
            5
          </option>
          <option
            value={10}
          >
            10
          </option>
          <option
            value={20}
          >
            20
          </option>
          <option
            value={50}
          >
            50
          </option>
        </select>
        <div
          className="form-switch"
        >
          <input
            checked={false}
            className="form-check-input"
            disabled={false}
            id={32}
            onChange={[Function]}
            type="checkbox"
          />
        </div>
      </div>
    </div>
  </div>,
]
`;

exports[`FeatureList FeaureList Initialized 1`] = `
Array [
  <p>
    General
  </p>,
  <div
    className="Grid"
  >
    <div
      className="Feature"
    >
      <label>
        <span
          className="GreyBox"
        />
        <span>
          CASE MANAGEMENT
        </span>
      </label>
      <div
        className="ToggleWithMax"
      >
        <div
          className="form-switch"
        >
          <input
            checked={false}
            className="form-check-input"
            disabled={false}
            id={11}
            onChange={[Function]}
            type="checkbox"
          />
        </div>
      </div>
    </div>
    <div
      className="Feature"
    >
      <label>
        <span
          className="GreyBox"
        />
        <span>
          MAP TIMELINE
        </span>
      </label>
      <div
        className="ToggleWithMax"
      >
        <div
          className="form-switch"
        >
          <input
            checked={false}
            className="form-check-input"
            disabled={false}
            id={12}
            onChange={[Function]}
            type="checkbox"
          />
        </div>
      </div>
    </div>
    <div
      className="Feature"
    >
      <label>
        <span
          className="GreyBox"
        />
        <span>
          VIEWS AND BRIEFINGS
        </span>
      </label>
      <div
        className="ToggleWithMax"
      >
        <div
          className="form-switch"
        >
          <input
            checked={false}
            className="form-check-input"
            disabled={false}
            id={13}
            onChange={[Function]}
            type="checkbox"
          />
        </div>
      </div>
    </div>
    <div
      className="Feature"
    >
      <label>
        <span
          className="GreyBox"
        />
        <span>
          NOTIFICATIONS
        </span>
      </label>
      <div
        className="ToggleWithMax"
      >
        <div
          className="form-switch"
        >
          <input
            checked={false}
            className="form-check-input"
            disabled={false}
            id={14}
            onChange={[Function]}
            type="checkbox"
          />
        </div>
      </div>
    </div>
    <div
      className="Feature"
    >
      <label>
        <span
          className="GreyBox"
        />
        <span>
          MASS COMMUNICATION
        </span>
      </label>
      <div
        className="ToggleWithMax"
      >
        <div
          className="form-switch"
        >
          <input
            checked={false}
            className="form-check-input"
            disabled={false}
            id={15}
            onChange={[Function]}
            type="checkbox"
          />
        </div>
      </div>
    </div>
    <div
      className="Feature"
    >
      <label>
        <span
          className="GreyBox"
        />
        <span>
          TRAFIC CAMERAS
        </span>
      </label>
      <div
        className="ToggleWithMax"
      >
        <div
          className="form-switch"
        >
          <input
            checked={false}
            className="form-check-input"
            disabled={false}
            id={16}
            onChange={[Function]}
            type="checkbox"
          />
        </div>
      </div>
    </div>
  </div>,
  <p>
    Settings
  </p>,
  <div
    className={null}
  >
    <div
      className="Feature"
    >
      <label>
        <span
          className="GreyBox"
        />
        <span>
          AUDIT LOG
        </span>
      </label>
      <div
        className="ToggleWithMax"
      >
        <div
          className="form-switch"
        >
          <input
            checked={false}
            className="form-check-input"
            disabled={false}
            id={21}
            onChange={[Function]}
            type="checkbox"
          />
        </div>
      </div>
    </div>
    <div
      className="accordion-item"
    >
      <div
        className="AccordianTitle"
      >
        <div
          className="FeatureAccordionHeader"
        >
          <label>
            <span
              className="GreyBox"
            />
            <span>
              USERS
            </span>
          </label>
          <div
            className="ToggleWithMax"
          >
            <div
              className="form-switch"
            >
              <input
                checked={false}
                className="form-check-input"
                disabled={false}
                id={22}
                onChange={[Function]}
                type="checkbox"
              />
            </div>
          </div>
        </div>
        <div
          className="toggle"
          onClick={[Function]}
        >
          <span
            className="UpCaret"
          />
        </div>
      </div>
    </div>
  </div>,
  <p>
    Alerts
  </p>,
  <div
    className={null}
  >
    <div
      className="Feature"
    >
      <label>
        <span
          className="GreyBox"
        />
        <span>
          ALERTS MANAGER
        </span>
      </label>
      <div
        className="ToggleWithMax"
      >
        <div
          className="form-switch"
        >
          <input
            checked={false}
            className="form-check-input"
            disabled={false}
            id={31}
            onChange={[Function]}
            type="checkbox"
          />
        </div>
      </div>
    </div>
    <div
      className="Feature"
    >
      <label>
        <span
          className="GreyBox"
        />
        <span>
          ALERT RULES
        </span>
      </label>
      <div
        className="ToggleWithMax"
      >
        <select
          className="Select"
          onChange={[Function]}
          value={10}
        >
          <option
            value={5}
          >
            5
          </option>
          <option
            value={10}
          >
            10
          </option>
          <option
            value={20}
          >
            20
          </option>
          <option
            value={50}
          >
            50
          </option>
        </select>
        <div
          className="form-switch"
        >
          <input
            checked={false}
            className="form-check-input"
            disabled={false}
            id={32}
            onChange={[Function]}
            type="checkbox"
          />
        </div>
      </div>
    </div>
  </div>,
]
`;

exports[`FeatureList FeaureList: Value Change 1`] = `
Array [
  <p>
    General
  </p>,
  <div
    className="Grid"
  >
    <div
      className="Feature"
    >
      <label>
        <span
          className="GreyBox"
        />
        <span>
          CASE MANAGEMENT
        </span>
      </label>
      <div
        className="ToggleWithMax"
      >
        <div
          className="form-switch"
        >
          <input
            checked={true}
            className="form-check-input"
            disabled={false}
            id={11}
            onChange={[Function]}
            type="checkbox"
          />
        </div>
      </div>
    </div>
    <div
      className="Feature"
    >
      <label>
        <span
          className="GreyBox"
        />
        <span>
          MAP TIMELINE
        </span>
      </label>
      <div
        className="ToggleWithMax"
      >
        <div
          className="form-switch"
        >
          <input
            checked={false}
            className="form-check-input"
            disabled={false}
            id={12}
            onChange={[Function]}
            type="checkbox"
          />
        </div>
      </div>
    </div>
    <div
      className="Feature"
    >
      <label>
        <span
          className="GreyBox"
        />
        <span>
          VIEWS AND BRIEFINGS
        </span>
      </label>
      <div
        className="ToggleWithMax"
      >
        <div
          className="form-switch"
        >
          <input
            checked={false}
            className="form-check-input"
            disabled={false}
            id={13}
            onChange={[Function]}
            type="checkbox"
          />
        </div>
      </div>
    </div>
    <div
      className="Feature"
    >
      <label>
        <span
          className="GreyBox"
        />
        <span>
          NOTIFICATIONS
        </span>
      </label>
      <div
        className="ToggleWithMax"
      >
        <div
          className="form-switch"
        >
          <input
            checked={false}
            className="form-check-input"
            disabled={false}
            id={14}
            onChange={[Function]}
            type="checkbox"
          />
        </div>
      </div>
    </div>
    <div
      className="Feature"
    >
      <label>
        <span
          className="GreyBox"
        />
        <span>
          MASS COMMUNICATION
        </span>
      </label>
      <div
        className="ToggleWithMax"
      >
        <div
          className="form-switch"
        >
          <input
            checked={false}
            className="form-check-input"
            disabled={false}
            id={15}
            onChange={[Function]}
            type="checkbox"
          />
        </div>
      </div>
    </div>
    <div
      className="Feature"
    >
      <label>
        <span
          className="GreyBox"
        />
        <span>
          TRAFIC CAMERAS
        </span>
      </label>
      <div
        className="ToggleWithMax"
      >
        <div
          className="form-switch"
        >
          <input
            checked={false}
            className="form-check-input"
            disabled={false}
            id={16}
            onChange={[Function]}
            type="checkbox"
          />
        </div>
      </div>
    </div>
  </div>,
  <p>
    Settings
  </p>,
  <div
    className={null}
  >
    <div
      className="Feature"
    >
      <label>
        <span
          className="GreyBox"
        />
        <span>
          AUDIT LOG
        </span>
      </label>
      <div
        className="ToggleWithMax"
      >
        <div
          className="form-switch"
        >
          <input
            checked={false}
            className="form-check-input"
            disabled={false}
            id={21}
            onChange={[Function]}
            type="checkbox"
          />
        </div>
      </div>
    </div>
    <div
      className="accordion-item"
    >
      <div
        className="AccordianTitle"
      >
        <div
          className="FeatureAccordionHeader"
        >
          <label>
            <span
              className="GreyBox"
            />
            <span>
              USERS
            </span>
          </label>
          <div
            className="ToggleWithMax"
          >
            <div
              className="form-switch"
            >
              <input
                checked={false}
                className="form-check-input"
                disabled={false}
                id={22}
                onChange={[Function]}
                type="checkbox"
              />
            </div>
          </div>
        </div>
        <div
          className="toggle"
          onClick={[Function]}
        >
          <span
            className="UpCaret"
          />
        </div>
      </div>
    </div>
  </div>,
  <p>
    Alerts
  </p>,
  <div
    className={null}
  >
    <div
      className="Feature"
    >
      <label>
        <span
          className="GreyBox"
        />
        <span>
          ALERTS MANAGER
        </span>
      </label>
      <div
        className="ToggleWithMax"
      >
        <div
          className="form-switch"
        >
          <input
            checked={false}
            className="form-check-input"
            disabled={false}
            id={31}
            onChange={[Function]}
            type="checkbox"
          />
        </div>
      </div>
    </div>
    <div
      className="Feature"
    >
      <label>
        <span
          className="GreyBox"
        />
        <span>
          ALERT RULES
        </span>
      </label>
      <div
        className="ToggleWithMax"
      >
        <select
          className="Select"
          onChange={[Function]}
          value={5}
        >
          <option
            value={5}
          >
            5
          </option>
          <option
            value={10}
          >
            10
          </option>
          <option
            value={20}
          >
            20
          </option>
          <option
            value={50}
          >
            50
          </option>
        </select>
        <div
          className="form-switch"
        >
          <input
            checked={false}
            className="form-check-input"
            disabled={false}
            id={32}
            onChange={[Function]}
            type="checkbox"
          />
        </div>
      </div>
    </div>
  </div>,
]
`;

exports[`FeatureList FeaureList:Individual:On Toggle Click 1`] = `
Array [
  <p>
    General
  </p>,
  <div
    className="Grid"
  >
    <div
      className="Feature"
    >
      <label>
        <span
          className="GreyBox"
        />
        <span>
          CASE MANAGEMENT
        </span>
      </label>
      <div
        className="ToggleWithMax"
      >
        <div
          className="form-switch"
        >
          <input
            checked={true}
            className="form-check-input"
            disabled={false}
            id={11}
            onChange={[Function]}
            type="checkbox"
          />
        </div>
      </div>
    </div>
    <div
      className="Feature"
    >
      <label>
        <span
          className="GreyBox"
        />
        <span>
          MAP TIMELINE
        </span>
      </label>
      <div
        className="ToggleWithMax"
      >
        <div
          className="form-switch"
        >
          <input
            checked={false}
            className="form-check-input"
            disabled={false}
            id={12}
            onChange={[Function]}
            type="checkbox"
          />
        </div>
      </div>
    </div>
    <div
      className="Feature"
    >
      <label>
        <span
          className="GreyBox"
        />
        <span>
          VIEWS AND BRIEFINGS
        </span>
      </label>
      <div
        className="ToggleWithMax"
      >
        <div
          className="form-switch"
        >
          <input
            checked={false}
            className="form-check-input"
            disabled={false}
            id={13}
            onChange={[Function]}
            type="checkbox"
          />
        </div>
      </div>
    </div>
    <div
      className="Feature"
    >
      <label>
        <span
          className="GreyBox"
        />
        <span>
          NOTIFICATIONS
        </span>
      </label>
      <div
        className="ToggleWithMax"
      >
        <div
          className="form-switch"
        >
          <input
            checked={false}
            className="form-check-input"
            disabled={false}
            id={14}
            onChange={[Function]}
            type="checkbox"
          />
        </div>
      </div>
    </div>
    <div
      className="Feature"
    >
      <label>
        <span
          className="GreyBox"
        />
        <span>
          MASS COMMUNICATION
        </span>
      </label>
      <div
        className="ToggleWithMax"
      >
        <div
          className="form-switch"
        >
          <input
            checked={false}
            className="form-check-input"
            disabled={false}
            id={15}
            onChange={[Function]}
            type="checkbox"
          />
        </div>
      </div>
    </div>
    <div
      className="Feature"
    >
      <label>
        <span
          className="GreyBox"
        />
        <span>
          TRAFIC CAMERAS
        </span>
      </label>
      <div
        className="ToggleWithMax"
      >
        <div
          className="form-switch"
        >
          <input
            checked={false}
            className="form-check-input"
            disabled={false}
            id={16}
            onChange={[Function]}
            type="checkbox"
          />
        </div>
      </div>
    </div>
  </div>,
  <p>
    Settings
  </p>,
  <div
    className={null}
  >
    <div
      className="Feature"
    >
      <label>
        <span
          className="GreyBox"
        />
        <span>
          AUDIT LOG
        </span>
      </label>
      <div
        className="ToggleWithMax"
      >
        <div
          className="form-switch"
        >
          <input
            checked={false}
            className="form-check-input"
            disabled={false}
            id={21}
            onChange={[Function]}
            type="checkbox"
          />
        </div>
      </div>
    </div>
    <div
      className="accordion-item"
    >
      <div
        className="AccordianTitle"
      >
        <div
          className="FeatureAccordionHeader"
        >
          <label>
            <span
              className="GreyBox"
            />
            <span>
              USERS
            </span>
          </label>
          <div
            className="ToggleWithMax"
          >
            <div
              className="form-switch"
            >
              <input
                checked={false}
                className="form-check-input"
                disabled={false}
                id={22}
                onChange={[Function]}
                type="checkbox"
              />
            </div>
          </div>
        </div>
        <div
          className="toggle"
          onClick={[Function]}
        >
          <span
            className="UpCaret"
          />
        </div>
      </div>
    </div>
  </div>,
  <p>
    Alerts
  </p>,
  <div
    className={null}
  >
    <div
      className="Feature"
    >
      <label>
        <span
          className="GreyBox"
        />
        <span>
          ALERTS MANAGER
        </span>
      </label>
      <div
        className="ToggleWithMax"
      >
        <div
          className="form-switch"
        >
          <input
            checked={false}
            className="form-check-input"
            disabled={false}
            id={31}
            onChange={[Function]}
            type="checkbox"
          />
        </div>
      </div>
    </div>
    <div
      className="Feature"
    >
      <label>
        <span
          className="GreyBox"
        />
        <span>
          ALERT RULES
        </span>
      </label>
      <div
        className="ToggleWithMax"
      >
        <select
          className="Select"
          onChange={[Function]}
          value={10}
        >
          <option
            value={5}
          >
            5
          </option>
          <option
            value={10}
          >
            10
          </option>
          <option
            value={20}
          >
            20
          </option>
          <option
            value={50}
          >
            50
          </option>
        </select>
        <div
          className="form-switch"
        >
          <input
            checked={false}
            className="form-check-input"
            disabled={false}
            id={32}
            onChange={[Function]}
            type="checkbox"
          />
        </div>
      </div>
    </div>
  </div>,
]
`;
